packages/server/lib/cloud/api.ts

```diff
@@ -384,38 +384,35 @@ module.exports = {
       })
     })
     .then(async (result: CreateRunResponse) => {
-      let protocolManager = new ProtocolManager()
+      const protocolManager = new ProtocolManager()
 
       const captureProtocolUrl = result.capture?.url || result.captureProtocolUrl
 
+      options.project.protocolManager = protocolManager
+
       debugProtocol({ captureProtocolUrl })
+
+      let script
+
       try {
         if (captureProtocolUrl || process.env.CYPRESS_LOCAL_PROTOCOL_PATH) {
-          const script = await this.getCaptureProtocolScript(captureProtocolUrl || process.env.CYPRESS_LOCAL_PROTOCOL_PATH)
-
-          if (script) {
-            options.project.protocolManager = protocolManager
-            await options.project.protocolManager.setupProtocol(script, result.runId)
-          }
+          script = await this.getCaptureProtocolScript(captureProtocolUrl || process.env.CYPRESS_LOCAL_PROTOCOL_PATH)
         }
       } catch (e) {
+        debugProtocol('Error downloading capture code', e)
+        const error = new Error(`Error downloading capture code: ${e.message}`)
+
         if (CAPTURE_ERRORS) {
-          await protocolManager.sendErrors([
-            {
-              args: [result.captureProtocolUrl],
-              captureMethod: 'getCaptureProtocolScript',
-              error: {
-                message: e.message,
-                stack: e.stack,
-                name: e.name,
-              },
-            },
-          ])
+          protocolManager.addFatalError('getCaptureProtocolScript', error, [result.captureProtocolUrl])
         } else {
           throw e
         }
       }
 
+      if (script) {
+        await options.project.protocolManager.setupProtocol(script, result.runId)
```

Comment:
What if `setupProtocol` throws? We should treat that as a fatal error as well right?

Reply:
Actually - setupProtocol is only in the app side of protocol manager; the constructor error state tests this case

Reply:
I see now 👍🏻